### PR TITLE
GeoJson Clustering

### PR DESCRIPTION
**Pull request by theduckylittle**
Add support for MB GeoJson Clustering.

### CODE REVIEW

File: src/components/map.jsx [hunk 1]
```diff
@@ -120,15 +134,27 @@ function updateGeojsonSource(olSource, glSource) {
...
   const vector_src = new VectorSource({
-    useSpatialIndex: false,
+    useSpatialIndex: true,
     wrapX: false,
   });
 
-  // see the vector source with the first update
-  //  before returning it.
-  updateGeojsonSource(vector_src, glSource);
+  // GeoJson sources can be clustered but OpenLayers
```
Comment (wnordmann) [inline src/components/map.jsx]: Very OL specific, consider moving to a olHelper.js in the future or some abstraction between OL and SDK

File: examples/clustering/config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+const config = {
+  mbx_key: 'pk.eyJ1IjoiZGJzcGF0aWFsIiwiYSI6IkRlbmZrOVEifQ.sbFFb6cfs8YWKDBeWRX4sw',
```
Comment (wnordmann) [inline examples/clustering/config.js]: can this key be kept in a public git repo?

Comment (theduckylittle) [inline examples/clustering/config.js]: Time to invalidate that key... that was added accidentally.

File: src/components/map.jsx (hunk 1 above)
Comment (theduckylittle) [inline src/components/map.jsx]: I'll address this in the other PR.

File: src/action-types.js [hunk 3]
```diff
@@ -13,6 +13,8 @@ export const MAP = {
   SET_LAYER_VISIBILITY: 'MAP_SET_LAYER_VISIBILITY',
   RECEIVE_CONTEXT: 'MAP_RECEIVE_CONTEXT',
   ORDER_LAYER: 'MAP_ORDER_LAYER',
+  CLUSTER_POINTS: 'MAP_CLUSTER_POINTS',
+  SET_CLUSTER_RADIUS: 'MAP_CLUSTER_RAIDUS',
```
Comment (bartvde) [inline src/action-types.js]: typo should be ```MAP_CLUSTER_RADIUS```

Comment (bartvde) [inline src/action-types.js]: probably should also include SET so ```MAP_SET_CLUSTER_RADIUS```

**Pull request merged**